homeassistant/components/verisure/climate.py

```diff
@@ -0,0 +1,218 @@
+"""Support for Verisure heatpump."""
+import logging
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    SUPPORT_TARGET_TEMPERATURE,
+    SUPPORT_FAN_MODE, SUPPORT_OPERATION_MODE,
+    SUPPORT_SWING_MODE, SUPPORT_ON_OFF)
+from homeassistant.const import TEMP_CELSIUS, ATTR_TEMPERATURE
+from homeassistant.components.verisure import HUB as hub
+from homeassistant.components.verisure import CONF_CLIMATE
+
+_LOGGER = logging.getLogger(__name__)
+HEAT_PUMPS = None
+VERISIRE_HASS_OP_MODE = {
+    'AUTO': 'auto',
+    'FAN': 'fan_only',
+    'COOL': 'cool',
+    'DRY': 'dry',
+    'HEAT': 'heat'
+}
+
+HASS_VERISURE_OP_MODE = {
+    'auto': 'AUTO',
+    'fan_only': 'FAN',
+    'cool': 'COOL',
+    'dry': 'DRY',
+    'heat': 'HEAT'
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Verisure heatpump."""
+    import jsonpath
+    jsonpath = jsonpath.jsonpath
+    global HEAT_PUMPS
+    hub.update_overview()
+    if int(hub.config.get(CONF_CLIMATE, 1)):
+        HEAT_PUMPS = hub.get('$.heatPumps')
+        if HEAT_PUMPS:
+            for heat_pump in HEAT_PUMPS[0]:
+                device_label = jsonpath(heat_pump, '$.deviceLabel')[0]
+                add_entities([
+                    VerisureHeatPump(device_label)
+                ])
+
+
+class VerisureHeatPump(ClimateDevice):
+    """Representation of a Verisure Heatpump."""
+
+    def __init__(self, heatpumpid):
+        """Initialize the climate device."""
+        import jsonpath
+        self.jsonpath = jsonpath.jsonpath
+        self._target_temperature = None
+        self._current_operation = None
+        self._current_fan_mode = None
+        self._current_swing_mode = None
+        self._on = None
+        self.heatpump_id = heatpumpid
+        self._support_flags = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE |\
+            SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF | SUPPORT_SWING_MODE
+        self._unit_of_measurement = TEMP_CELSIUS
+        self._fan_list = ['Auto', 'Low', 'Medium_Low', 'Medium',
+                          'Medium_High', 'High']
+        self._operation_list = ['heat', 'cool', 'auto', 'dry', 'fan_only']
+        self._swing_list = ['Auto', '0_Degrees', '30_Degrees', '60_Degrees',
+                            '90_Degrees']
+        self._config_date = None
+        self.sync_data()
```

Comment:
Remove this. Instead call `add_entities` with second argument `True`. This will make sure that `update` method is called during entity addition.